api/src/identity-access-management/scripts/delete-ex-sco-authentication-methods.script.js

```diff
@@ -0,0 +1,66 @@
+import { knex } from '../../../db/knex-database-connection.js';
+import { Script } from '../../shared/application/scripts/script.js';
+import { ScriptRunner } from '../../shared/application/scripts/script-runner.js';
+import { NON_OIDC_IDENTITY_PROVIDERS } from '../domain/constants/identity-providers.js';
+import { removeByUserIdAndIdentityProvider } from '../infrastructure/repositories/authentication-method.repository.js';
+import { updateUsername } from '../infrastructure/repositories/user.repository.js';
+
+export class DeleteExScoAuthenticationMethodsScript extends Script {
+  constructor() {
+    super({
+      description:
+        'Deletes authentication methods for users that are no longer SCO and that have already used the account recovery demand.',
+      permanent: false,
+      options: {
+        dryRun: {
+          type: 'boolean',
+          describe: 'If present, the script will not make any changes to the database',
+          default: false,
+        },
+      },
+    });
+  }
+
+  async handle({ options, logger, dependencies = { removeByUserIdAndIdentityProvider, updateUsername } }) {
```

Comment:
Dans tout le reste du code de Pix quand on définit ainsi un argument `dependencies` c'est pour passer un repository, un service et non une fonction de repository ou de service. Aussi voici une suggestion dans ce sens : 

```suggestion
  async handle({ options, logger, dependencies = { authenticationMethodRepository, userRepository } }) {
```